projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/ExecutionDataFetcher.java

```diff
@@ -305,4 +308,52 @@ private static DataJobPage buildResponse(
             .totalPages(pageSize)
             .totalItems(count).build();
    }
+
+   List<V2DataJob> populateStatusCounts(List<V2DataJob> dataJos, DataFetchingEnvironment dataFetchingEnvironment) {
+
+      List<ExecutionStatus> statusesToCount = determineStatusesToCount(dataFetchingEnvironment);
+      List<String> jobsList = dataJos.stream().map(V2DataJob::getJobName).collect(Collectors.toList());
+      Map<String, Map<ExecutionStatus, Integer>> response = jobExecutionService.countExecutionStatuses(jobsList, statusesToCount);
+
+      dataJos.stream()
+              .forEach(job -> {
+                 if (job.getDeployments() != null) {
+                    job.getDeployments()
+                            .stream()
+                            .findFirst()
+                            .ifPresent(deployment -> {
+                               setStatusCounts(response, job, dataFetchingEnvironment, deployment);
+                            });
+                 }
+              });
+      return dataJos;
+   }
+
+   private void setStatusCounts(Map<String, Map<ExecutionStatus, Integer>> response, V2DataJob job,
+                              DataFetchingEnvironment dataFetchingEnvironment, V2DataJobDeployment v2DataJobDeployment) {
+
+      DataFetchingFieldSelectionSet selectionSet = dataFetchingEnvironment.getSelectionSet();
+      Map<ExecutionStatus, Integer> statusCountsPerJob = response.getOrDefault(job.getJobName(), Map.of());
+
+      if (selectionSet.contains(JobFieldStrategyBy.DEPLOYMENT_FAILED_EXECUTIONS.getPath())) {
+         v2DataJobDeployment.setFailedExecutions(statusCountsPerJob.getOrDefault(ExecutionStatus.FAILED, 0));
+      }
+
+      if (selectionSet.contains(JobFieldStrategyBy.DEPLOYMENT_SUCCESSFUL_EXECUTIONS.getPath())) {
+         v2DataJobDeployment.setSuccessfulExecutions(statusCountsPerJob.getOrDefault(ExecutionStatus.FINISHED, 0));
+      }
+   }
+
+   private List<ExecutionStatus> determineStatusesToCount(DataFetchingEnvironment dataFetchingEnvironment) {
+      DataFetchingFieldSelectionSet selectionSet = dataFetchingEnvironment.getSelectionSet();
+      List<ExecutionStatus> statusesToCount = new ArrayList<>();
+      if (selectionSet.contains(JobFieldStrategyBy.DEPLOYMENT_FAILED_EXECUTIONS.getPath())) {
```

Comment:
I am wondering why are you using two different code styles for this method and the above one?

Reply:
What do you mean?

Reply:
One method has a blank line before `if` statement and the other one does not.

Reply:
I never did put any thought into it, probably felt visually better that way when I wrote the methods. Most coding standards I've read say that blank lines may (or may not appear) anywhere where they improve readability and either option is fine. Is this a real issue though?

Reply:
just be consistent :)

Reply:
Both methods determineStatusesToCount and setStatusCounts have been added in this change. But both have different code style.